### PR TITLE
Extremely slow proxing pages with 302 http status code (fix #390)

### LINKED ISSUE #390: Extremely slow proxing pages with 302 http status code
Open http://kinozal.tv/ in playground.
Open http://kinozal.tv/ without proxing.

With proxing page will loaded after 45 sec.
Without proxing - about 100 ms.

See video - http://screencast.com/t/BfrixaXWUM

### PR DESCRIPTION
**Pull request by LavrovArtem**
\cc @inikulin @churkin

### CODE REVIEW

File: src/request-pipeline/context.js [hunk 1]
```diff
@@ -136,18 +138,21 @@ export default class RequestPipelineContext {
     }
 
     buildContentInfo () {
-        var contentType = this.destRes.headers['content-type'] || '';
+        var destRes     = this.destRes;
```
Comment (inikulin) [inline src/request-pipeline/context.js]: No need for this var

File: src/request-pipeline/context.js [hunk 2]
```diff
@@ -136,18 +138,21 @@ export default class RequestPipelineContext {
...
+        var encoding    = destRes.headers['content-encoding'];
 
         var isCSS                   = contentTypeUtils.isCSSResource(contentType, accept);
         var isManifest              = contentTypeUtils.isManifest(contentType);
         var isScript                = this.dest.isScript || contentTypeUtils.isScriptResource(contentType, accept);
         var isForm                  = this.dest.isForm;
-        var isFormWithEmptyResponse = isForm && this.destRes.statusCode === 204;
+        var isFormWithEmptyResponse = isForm && destRes.statusCode === 204;
+
+        var isRedirect = REDIRECT_STATUS_CODES.indexOf(destRes.statusCode) > -1 && destRes.headers['location'];
```
Comment (inikulin) [inline src/request-pipeline/context.js]: Check for the location first, since it's more low-cost operation

**Comment by inikulin** [comment]
\r-

**Comment by LavrovArtem** [comment]
FPR

**Comment by inikulin** [comment]
lgtm

**Comment by inikulin** [comment]
Since, @churkin is not available at the moment, @AlexanderMoskovkin can you review instead, pls?

**Comment by AlexanderMoskovkin** [comment]
lgtm

**Pull request merged**